File: 0200-number-of-islands/0200-number-of-islands.cpp
class Solution {
public:
    int m,n;
    void dfs(vector<vector<char>>&grid, int r, int c){
        if(r < 0 || c < 0 || r >= m || c >= n || grid[r][c] == '0')
            return;
        grid[r][c] = '0';
        dfs(grid,r-1,c);
        dfs(grid,r,c-1);
        dfs(grid,r+1,c);
        dfs(grid,r,c+1);
    }
    int numIslands(vector<vector<char>>& grid) {
        m = grid.size();
        n = grid[0].size();
        int count = 0;
        for(int r = 0; r < m; r++){
            for(int c = 0; c < n; c++){
                if(grid[r][c] == '1'){
                    count++;
                    dfs(grid,r,c);
                }
            }
        }
        return count;

    }
};

File: 0200-number-of-islands/0200-number-of-islands.py
class Solution(object):
    
    def numIslands(self, grid):
        self.m = len(grid)
        self.n = len(grid[0])
        count = 0
        for i in range(0, self.m):
            for j in range(0,self.n):
                if grid[i][j] == '1':
                    count+=1
                    self.dfs(grid,i,j)
        return count
    def dfs(self,grid,r,c):
        if r < 0 or c < 0 or r >= self.m or c >= self.n or grid[r][c] == '0':
            return
        grid[r][c] = '0'
        self.dfs(grid,r-1,c)
        self.dfs(grid,r+1,c)
        self.dfs(grid,r,c-1)
        self.dfs(grid,r,c+1)
    
        